sdcm/tester.py

```diff
@@ -3161,21 +3161,29 @@ def check_specified_stats_regression(self, stats):
         except Exception as ex:  # pylint: disable=broad-except
             self.log.exception('Failed to check regression: %s', ex)
 
-    def wait_no_compactions_running(self, n=80, sleep_time=60):  # pylint: disable=invalid-name
-        # Wait for up to 80 mins that there are no running compactions
+    @property
+    def is_compaction_running(self):
+        compaction_query = "sum(scylla_compaction_manager_compactions{})"
+        now = time.time()
+        results = self.prometheus_db.query(query=compaction_query, start=now - 60, end=now)
+        self.log.debug("scylla_compaction_manager_compactions: {results}".format(**locals()))
+        assert results or results == [], "No results from Prometheus"
+        # if any result values is not zero - there are running compactions.
+        return any((float(v[1]) for v in results[0]["values"]))
+
+    def wait_compactions_are_running(self, n=20, sleep_time=60):  # pylint: disable=invalid-name
+        # Wait until there are running compactions
         @retrying(n=n, sleep_time=sleep_time, allowed_exceptions=(AssertionError,))
-        def is_compactions_done():
-            compaction_query = "sum(scylla_compaction_manager_compactions{})"
-            now = time.time()
-            results = self.prometheus_db.query(query=compaction_query, start=now - 60, end=now)
-            self.log.debug("scylla_compaction_manager_compactions: {results}".format(**locals()))
-            assert results or results == [], "No results from Prometheus"
-            # if all are zeros the result will be False, otherwise there are still compactions
-            if results:
-                assert any((float(v[1]) for v in results[0]["values"])) is False, \
-                    "Waiting until all compactions settle down"
+        def _is_compaction_running():
+            assert self.is_compaction_running, "Waiting for compaction to start"
+        _is_compaction_running()
 
-        is_compactions_done()
+    def wait_no_compactions_running(self, n=80, sleep_time=60):  # pylint: disable=invalid-name
+        # Wait until there are no running compactions
+        @retrying(n=n, sleep_time=sleep_time, allowed_exceptions=(AssertionError,))
+        def _is_no_compaction_running():
+            assert not self.is_compaction_running, "Waiting until all compactions settle down"
```

Comment:
in regular workloads it may be not right, as if we continue running writes, we may have other compactions running... did you check all calls of this function to be sure it does make sense to check if `no compaction running`?
most of calls are on performance jobs BTW

Reply:
@fgelcer , the idea is just making a rewrite to this method. I think no logic changes done. We can consider running any other job that uses this method in order to be on the safe side.